### ISSUE TITLE
Assignability of valid value to union of builtin subtypes of int depends on the order of definition 

### ISSUE THREAD

**Issue by suleka96** [ballerina-platform/ballerina-lang#33689]
**Description:**
$title

**Steps to reproduce:**
```
import ballerina/io;

public function main() {
    int:Signed8|int:Unsigned32 a1 = 235262; // works
    int:Unsigned32|int:Signed8 a2 = 235262; // incompatible types: expected 'int:Signed8', found 'int'
}
```

**Related Issues (optional):**
#32686

**Comment by chiranSachintha** [comment]
@suleka96 This is fixed in `Swan Lake Beta4 RC2`. Right?

**Comment by KavinduZoysa** [comment]
This is fixed in `Ballerina 2201.3.0`.
